src/pages/Popup/Popup.tsx

```diff
@@ -9,13 +9,19 @@ import { AppTab } from '../../types/tab.type';
 const Popup = () => {
   const [tabs, setTabs] = useState<AppTab[]>([]);
   const [repo, setRepo] = useState(getRepoNameFromStorage());
+  const [uniqueUrls, setUniqueUrls] = useState<Set<string>>(new Set());
 
   const getTabs = useCallback(async () => {
     const tabs = await chrome.tabs.query({
       url: ['https://github.com/*/pull/*/files'],
     }) as AppTab[];
     console.log({ tabs });
-    const filteredTabs = tabs.filter((tab: AppTab) => {
+    const filteredTabs = tabs
+    .filter((tab: AppTab) => {
+      if(uniqueUrls && uniqueUrls.has(tab.url!)){
+        return false;
+      }
+      setUniqueUrls(uniqueUrls.add(tab.url!));
```

Comment:
Pretty nice solution 🎉 
However, we can avoid putting this `uniqueUrls` as a state variable since we don't really use this in the UI, we just use the variable in the function and then it goes away, so we may just use it as:

```suggestion
    const uniqueUrls = new Set();
    const filteredTabs = tabs
    .filter((tab: AppTab) => {
      if(uniqueUrls && uniqueUrls.has(tab.url!)){
        return false;
      }
      setUniqueUrls(uniqueUrls.add(tab.url!));
```